torch/distributed/tensor/_random.py

```diff
@@ -170,7 +170,10 @@ def __init__(
         super().__init__(_resolve_device(device_mesh=device_mesh))
         assert self._device_handle is not None
         # DTensor RNG tracker so far only supports CUDA/CUDA-like devices
-        if self._device.type != "cuda":
+        if self._device.type != "cuda" and (
+            not hasattr(self._device_handle, "set_rng_state")
+            or not hasattr(self._device_handle, "get_rng_state")
+        ):
```

Comment:
I think this would not only allow cuda-like devices but also CPU. For example:
```
CUDA_VISIBLE_DEVICES="" python
Python 3.12.9 | packaged by Anaconda, Inc. | (main, Feb  6 2025, 18:56:27) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.cuda.is_available()
False
>>> hasattr(torch, "set_rng_state")
True
>>> 
```

Reply:
@XilunWu Thanks! Totally agree. I'm gonna make change.